src/LockTable.cc

```diff
@@ -0,0 +1,225 @@
+/* Copyright (c) 2014-2015 Stanford University
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIM ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL AUTHORS BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include "LockTable.h"
+#include "BitOps.h"
+#include "Memory.h"
+
+namespace RAMCloud {
+
+/**
+ * Constructor for LockTable.
+ *
+ * \param numBuckets
+ *      Locks in a LockTable are stored in one or more buckets.  This parameter
+ *      sets the number of buckets that should be used for this new LockTable.
+ *      This value should be a power of two.
```

Comment:
Parameterizing this with buckets is likely to be confusing to the caller. For example, the caller needs to know how many entries are in a bucket (plus the fact that the first and last entries aren't really entries) in order to figure out the right number of buckets to ask for. Perhaps it would be better for the argument to be the number of entries of storage (without overflows) that the caller would like, then you compute the right number of buckets based on that.